tests/test_live.py

```diff
@@ -0,0 +1,1428 @@
+                TEST_GCS_BUCKET, prefix), "gs://{}/{}/d3/d31/d312/f3132".format(
+                    TEST_GCS_BUCKET, prefix), "gs://{}/{}/d3/d32/f321".format(
+                        TEST_GCS_BUCKET, prefix),
+        "gs://{}/{}/play/d1/ff".format(
+            TEST_GCS_BUCKET, prefix), "gs://{}/{}/play/d1:/ff".format(
+                TEST_GCS_BUCKET, prefix), "gs://{}/{}/play/d2/ff".format(
+                    TEST_GCS_BUCKET, prefix), "gs://{}/{}/play/test1".format(
+                        TEST_GCS_BUCKET, prefix),
+        "gs://{}/{}/same_file_different_dir/d1/d11/d111/ff".format(
+            TEST_GCS_BUCKET,
+            prefix), "gs://{}/{}/same_file_different_dir/d1/d11/ff".format(
+                TEST_GCS_BUCKET,
+                prefix), "gs://{}/{}/same_file_different_dir/d1/ff".format(
+                    TEST_GCS_BUCKET,
+                    prefix), "gs://{}/{}/same_file_different_dir/d2/ff".format(
+                        TEST_GCS_BUCKET, prefix)
+    ]
+    futures = []  # type: List[concurrent.futures.Future]
+    with temppathlib.NamedTemporaryFile() as tmp_file:
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            tmp_file.path.write_text(GCS_FILE_CONTENT)
+            for file in gcs_structure:
+
+                cmd = [
+                    'gsutil', '-q', 'cp', '-n',
+                    tmp_file.path.as_posix(), file
+                ]
+
+                setup_thread = executor.submit(
+                    subprocess_single_cmd_execution, cmd=cmd, file=file)
+                futures.append(setup_thread)
+
+
+def subprocess_single_cmd_execution(cmd: List[str], file: str):
+    proc = subprocess.Popen(cmd, stdin=subprocess.PIPE)
+    proc.communicate()
+    if proc.returncode != 0:
+        raise RuntimeError("Failed to write to the object: {}".format(file))
+
+
+def gcs_test_teardown(prefix: str):
+    cmd = [
+        "gsutil", "-m", "rm", "-r", "gs://{}/{}".format(TEST_GCS_BUCKET, prefix)
+    ]
+
+    proc = subprocess.Popen(
+        cmd,
+        universal_newlines=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE)
+    proc.communicate()
+
+
+def call_gsutil_ls(path: str, recursive: bool = False) -> List[str]:
+    if recursive:
+        cmd = ["gsutil", "ls", "-r", path]
+    else:
+        cmd = ["gsutil", "ls", path]
+
+    proc = subprocess.Popen(
+        cmd,
+        universal_newlines=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE)
+    stdout, stderr = proc.communicate()
+
+    if proc.returncode != 0:
+        print(stderr)
+
+    lines = []  # type: List[str]
+    for line in stdout.split('\n'):
+        line = line.strip()
+        # empty line
+        if line == '':
+            continue
+        # subdirectory matching resolved wildcard *
+        if line.endswith('/:'):
+            continue
+        lines.append(line)
+
+    return lines
+
+
+def call_gcs_client_ls(client: gswrap.Client,
```

Comment:
What is this wrapper used for? I think it is not safe to catch the exceptions just like this